<4:2016/7/14 0:29:30:P2PChatServer.cs:393> RecvCb , readWriteObject.IsHeader: True
<4:2016/7/14 0:29:30:P2PChatServer.cs:393> RecvCb , readWriteObject.IsHeader: False
<4:2016/7/14 0:29:30:P2PChatServer.cs:421> RecvCb: eLogin; _server_; 100000; 1
<4:2016/7/14 0:29:30:ChatUser.cs:121> 100001
<4:2016/7/14 0:29:30:ChatUser.cs:121> 100002
<4:2016/7/14 0:29:30:P2PChatServer.cs:206> AsyncSend: eLoginSuccess; 100000; _server_; 1
<4:2016/7/14 0:29:30:P2PChatServer.cs:456> AsyncSendCb: 
<4:2016/7/14 0:29:38:P2PChatServer.cs:393> RecvCb , readWriteObject.IsHeader: True
<4:2016/7/14 0:29:38:P2PChatServer.cs:393> RecvCb , readWriteObject.IsHeader: False
<4:2016/7/14 0:29:38:P2PChatServer.cs:421> RecvCb: eLogin; _server_; 100001; 2
<4:2016/7/14 0:29:38:ChatUser.cs:121> 100000
<4:2016/7/14 0:29:38:ChatUser.cs:121> 100002
<4:2016/7/14 0:29:38:P2PChatServer.cs:206> AsyncSend: eLoginSuccess; 100001; _server_; 2
<4:2016/7/14 0:29:38:P2PChatServer.cs:456> AsyncSendCb: 
<4:2016/7/14 0:29:46:P2PChatServer.cs:393> RecvCb , readWriteObject.IsHeader: True
<4:2016/7/14 0:29:46:P2PChatServer.cs:393> RecvCb , readWriteObject.IsHeader: False
<4:2016/7/14 0:29:46:P2PChatServer.cs:421> RecvCb: eLogin; _server_; 100002; 4
<4:2016/7/14 0:29:46:ChatUser.cs:121> 100000
<4:2016/7/14 0:29:46:ChatUser.cs:121> 100001
<4:2016/7/14 0:29:46:P2PChatServer.cs:206> AsyncSend: eLoginSuccess; 100002; _server_; 3
<4:2016/7/14 0:29:46:P2PChatServer.cs:456> AsyncSendCb: 
<4:2016/7/14 0:30:21:P2PChatServer.cs:393> RecvCb , readWriteObject.IsHeader: True
失败: 
<4:2016/7/14 0:30:22:P2PChatServer.cs:393> RecvCb , readWriteObject.IsHeader: True
失败: 
<4:2016/7/14 0:30:24:P2PChatServer.cs:393> RecvCb , readWriteObject.IsHeader: True
失败: 

********************* error *********************
  DateTime : 2016/7/14 0:30:24
  FileName : P2PChatServer.cs
LineNumber : 442
  Error : System.IO.IOException: 无法从传输连接中读取数据: 远程主机强迫关闭了一个现有的连接。。 ---> System.Net.Sockets.SocketException: 远程主机强迫关闭了一个现有的连接。
   在 System.Net.Sockets.Socket.EndReceive(IAsyncResult asyncResult)
   在 System.Net.Sockets.NetworkStream.EndRead(IAsyncResult asyncResult)
   --- 内部异常堆栈跟踪的结尾 ---
   在 System.Net.Sockets.NetworkStream.EndRead(IAsyncResult asyncResult)
   在 P2PService.P2PChatServer.RecveCb(IAsyncResult iar) 位置 C:\Users\pzh\Desktop\JobAndTask\P2PService\P2PChatServer.cs:行号 377

********************* error *********************

<4:2016/7/14 0:30:24:P2PChatServer.cs:206> AsyncSend: eOffline; ; 100002; 4

********************* error *********************
  DateTime : 2016/7/14 0:30:24
  FileName : P2PChatServer.cs
LineNumber : 233
  Error : System.IO.IOException: 无法将数据写入传输连接: 远程主机强迫关闭了一个现有的连接。。 ---> System.Net.Sockets.SocketException: 远程主机强迫关闭了一个现有的连接。
   在 System.Net.Sockets.Socket.BeginSend(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags, AsyncCallback callback, Object state)
   在 System.Net.Sockets.NetworkStream.BeginWrite(Byte[] buffer, Int32 offset, Int32 size, AsyncCallback callback, Object state)
   --- 内部异常堆栈跟踪的结尾 ---
   在 System.Net.Sockets.NetworkStream.BeginWrite(Byte[] buffer, Int32 offset, Int32 size, AsyncCallback callback, Object state)
   在 P2PService.P2PChatServer.AsyncSend(BaseTrans trans, TcpClient client) 位置 C:\Users\pzh\Desktop\JobAndTask\P2PService\P2PChatServer.cs:行号 218

********************* error *********************

<4:2016/7/14 0:30:24:P2PChatServer.cs:206> AsyncSend: eOffline; ; 100001; 5

********************* error *********************
  DateTime : 2016/7/14 0:30:24
  FileName : P2PChatServer.cs
LineNumber : 233
  Error : System.IO.IOException: 无法将数据写入传输连接: 远程主机强迫关闭了一个现有的连接。。 ---> System.Net.Sockets.SocketException: 远程主机强迫关闭了一个现有的连接。
   在 System.Net.Sockets.Socket.BeginSend(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags, AsyncCallback callback, Object state)
   在 System.Net.Sockets.NetworkStream.BeginWrite(Byte[] buffer, Int32 offset, Int32 size, AsyncCallback callback, Object state)
   --- 内部异常堆栈跟踪的结尾 ---
   在 System.Net.Sockets.NetworkStream.BeginWrite(Byte[] buffer, Int32 offset, Int32 size, AsyncCallback callback, Object state)
   在 P2PService.P2PChatServer.AsyncSend(BaseTrans trans, TcpClient client) 位置 C:\Users\pzh\Desktop\JobAndTask\P2PService\P2PChatServer.cs:行号 218

********************* error *********************


********************* error *********************
  DateTime : 2016/7/14 0:30:24
  FileName : P2PChatServer.cs
LineNumber : 321
  Error : System.InvalidOperationException: 集合已修改；可能无法执行枚举操作。
   在 System.Collections.Hashtable.HashtableEnumerator.MoveNext()
   在 ChatServerUI.ChatServerForm.SendOffline(String _userId) 位置 C:\Users\pzh\Desktop\JobAndTask\ChatServerUI\ChatServerForm.cs:行号 278
   在 ChatServerUI.ChatServerForm.ChatServerForm_TcpDisConnect(Object sender, TcpDisConnectEventArgs e) 位置 C:\Users\pzh\Desktop\JobAndTask\ChatServerUI\ChatServerForm.cs:行号 110
   在 P2PService.P2PChatServer.OnTcpDisConnect(TcpDisConnectEventArgs e) 位置 C:\Users\pzh\Desktop\JobAndTask\P2PService\P2PChatServer.cs:行号 77
   在 P2PService.P2PChatServer.DisConnectClient(ReadWriteObject readWriteObject) 位置 C:\Users\pzh\Desktop\JobAndTask\P2PService\P2PChatServer.cs:行号 311

********************* error *********************


********************* error *********************
  DateTime : 2016/7/14 0:30:24
  FileName : P2PChatServer.cs
LineNumber : 321
  Error : System.InvalidOperationException: 集合已修改；可能无法执行枚举操作。
   在 System.Collections.Hashtable.HashtableEnumerator.MoveNext()
   在 ChatServerUI.ChatServerForm.SendOffline(String _userId) 位置 C:\Users\pzh\Desktop\JobAndTask\ChatServerUI\ChatServerForm.cs:行号 278
   在 ChatServerUI.ChatServerForm.ChatServerForm_TcpDisConnect(Object sender, TcpDisConnectEventArgs e) 位置 C:\Users\pzh\Desktop\JobAndTask\ChatServerUI\ChatServerForm.cs:行号 110
   在 P2PService.P2PChatServer.OnTcpDisConnect(TcpDisConnectEventArgs e) 位置 C:\Users\pzh\Desktop\JobAndTask\P2PService\P2PChatServer.cs:行号 77
   在 P2PService.P2PChatServer.DisConnectClient(ReadWriteObject readWriteObject) 位置 C:\Users\pzh\Desktop\JobAndTask\P2PService\P2PChatServer.cs:行号 311

********************* error *********************


********************* error *********************
  DateTime : 2016/7/14 0:30:24
  FileName : P2PChatServer.cs
LineNumber : 442
  Error : System.IO.IOException: 无法从传输连接中读取数据: 远程主机强迫关闭了一个现有的连接。。 ---> System.Net.Sockets.SocketException: 远程主机强迫关闭了一个现有的连接。
   在 System.Net.Sockets.Socket.EndReceive(IAsyncResult asyncResult)
   在 System.Net.Sockets.NetworkStream.EndRead(IAsyncResult asyncResult)
   --- 内部异常堆栈跟踪的结尾 ---
   在 System.Net.Sockets.NetworkStream.EndRead(IAsyncResult asyncResult)
   在 P2PService.P2PChatServer.RecveCb(IAsyncResult iar) 位置 C:\Users\pzh\Desktop\JobAndTask\P2PService\P2PChatServer.cs:行号 377

********************* error *********************


********************* error *********************
  DateTime : 2016/7/14 0:30:24
  FileName : P2PChatServer.cs
LineNumber : 442
  Error : System.IO.IOException: 无法从传输连接中读取数据: 远程主机强迫关闭了一个现有的连接。。 ---> System.Net.Sockets.SocketException: 远程主机强迫关闭了一个现有的连接。
   在 System.Net.Sockets.Socket.EndReceive(IAsyncResult asyncResult)
   在 System.Net.Sockets.NetworkStream.EndRead(IAsyncResult asyncResult)
   --- 内部异常堆栈跟踪的结尾 ---
   在 System.Net.Sockets.NetworkStream.EndRead(IAsyncResult asyncResult)
   在 P2PService.P2PChatServer.RecveCb(IAsyncResult iar) 位置 C:\Users\pzh\Desktop\JobAndTask\P2PService\P2PChatServer.cs:行号 377

********************* error *********************

<4:2016/7/14 0:30:24:P2PChatServer.cs:383> Alarm
<4:2016/7/14 0:30:24:P2PChatServer.cs:383> Alarm
<4:2016/7/14 0:30:24:P2PChatServer.cs:383> Alarm
